app/api/events_view.py

```diff
@@ -52,46 +74,78 @@ def read_events_from_file():
         string = string.replace("\n", "")
         jsonvar = json.loads(string)
         device, sim, app_version_code = set_events_context(jsonvar)
-        return read_and_save_events(jsonvar, device, sim, app_version_code)
+
+        if sim:
+            events = prepare_events(jsonvar, device.device_id, sim.serial_number, app_version_code)
+        else:
+            events = prepare_events(jsonvar, device.device_id, None, app_version_code)
 
     except json.JSONDecodeError as e:
         app.logger.error("JSONDecodeError: " + str(e))
+        return "Bad Request", 400
     except BadRequestKeyError as e:
         app.logger.error("BadRequestKeyError: " + str(e))
+        return "Bad Request", 400
     except UnicodeError as e:
         app.logger.error("UnicodeError: " + str(e))
+        return "Bad Request", 400
     except TypeError as e:
         app.logger.error("TypeError: " + str(e))
+        return "Bad Request", 400
     except KeyError as e:
         app.logger.error("KeyError: " + str(e))
-        db.session.rollback()
+        return "Bad Request", 400
     except Exception as e:
         app.logger.error("Unknow Exception: " + str(e))
-        db.session.rollback()
-    return "Bad Request", 400
+        return "Bad Request", 400
 
+    # continue with processation of events
+    return save_events(events)
 
-def read_and_save_events(jsonvar, device, sim, app_version_code):
-    total_events = 0
-    del jsonvar["device_records"]
-    del jsonvar["sim_records"]
 
-    for events_name, events in jsonvar.items():
-        total_events += add_events(events_name, events, device, sim, app_version_code)
+def save_events(events):
+    """
+    Save events to the database
+    :param events: List with the events to save
+    :return: HTML answer
+    """
+    for event in events:
+        db.session.add(event)
 
     try:
         # save and commit events and information to the database
         db.session.commit()
     except Exception as e:
-        app.logger.error("Error adding events to de database " + str(e))
+        app.logger.error("Error adding events to database " + str(e))
         db.session.rollback()
         return "Bad Request", 400
```

Comment:
Maybe could be better to return another code, not 400 (to differentiate this type of error with typeError, keyError, UnicodeError, etc)